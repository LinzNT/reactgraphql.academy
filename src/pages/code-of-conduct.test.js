import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import CodeOfConduct from './code-of-conduct'

describe('<CodeOfConduct />', () => {
  it('should render component', () => {
    const wrapper = shallow(<CodeOfConduct />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
