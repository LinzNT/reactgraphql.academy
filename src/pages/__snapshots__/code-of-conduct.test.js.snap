// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodeOfConduct /> should render component 1`] = `
<Fragment>
  <Header
    bgImg="about-us"
    links={
      Array [
        Object {
          "text": "Key takeaways ",
          "to": "key-takeaways",
        },
        Object {
          "text": "More detail",
          "to": "more-detail",
        },
      ]
    }
    subtitle="All students, coaches, sponsors and volunteers at our trainings are<br /> required to agree with the following code of conduct.            "
    titleLines={
      Array [
        "Our code of conduct",
      ]
    }
  />
  <TopSection>
    <Grid>
      <Card
        border="shadow"
      >
        <Row>
          <Styled(Col)
            lg={10}
            lgOffset={1}
          >
            <Row>
              <Styled(Col)
                md={5}
              >
                <Styled(styled.h2)>
                  Key takeways
                  <Link
                    name="key-takeaways"
                    to="#key-takeaways"
                  >
                    #
                  </Link>
                </Styled(styled.h2)>
                <styled.ul>
                  <styled.li>
                    Everyone is required to adhere to the code of conduct -
                     
                    <strong>
                      everyone
                    </strong>
                    .
                  </styled.li>
                  <styled.li>
                    ReactJS Academy is a harassment-free zone - offline AND online.
                  </styled.li>
                  <styled.li>
                    If you don’t adhere, you’ll get thrown out without a refund.
                  </styled.li>
                  <styled.li>
                    Basically, don’t be a dick.
                  </styled.li>
                </styled.ul>
                <styled.img
                  src="https://firebasestorage.googleapis.com/v0/b/reactjsacademy-react.appspot.com/o/codeofconduct_keypoints.jpg?alt=media"
                />
              </Styled(Col)>
              <Styled(Col)
                md={6}
                mdOffset={1}
              >
                <styled.h2>
                  Summary
                </styled.h2>
                <styled.p>
                  Every person at our trainings will be required to agree with this code of conduct and organisers will enforce this code throughout any event run by ReactJS Academy.
                </styled.p>
                <styled.p>
                  All participants are expected to cooperate with us to help guarentee a safe environment for everybody.
                </styled.p>
                <styled.p>
                  Our trainings are dedicated to providing a harassment-free training experience for all who attend, regardless of:
                </styled.p>
                <styled.ul>
                  <styled.li>
                    gender
                  </styled.li>
                  <styled.li>
                    gender identity and expression
                  </styled.li>
                  <styled.li>
                    age
                  </styled.li>
                  <styled.li>
                    sexual orientation
                  </styled.li>
                  <styled.li>
                    disability
                  </styled.li>
                  <styled.li>
                    physical appearance
                  </styled.li>
                  <styled.li>
                    body size
                  </styled.li>
                  <styled.li>
                    race
                  </styled.li>
                  <styled.li>
                    ethnicity
                  </styled.li>
                  <styled.li>
                    religion (or lack thereof)
                  </styled.li>
                  <styled.li>
                    technology choices
                  </styled.li>
                </styled.ul>
                <styled.p>
                  We will not accept any form of harassment of our participants. Sexual imagery and language is not appropriate at any of our events, including formal trainings, talks/meetups, workshops, or parties. This is also applicable through social media such as Twitter and other online outlouts.
                </styled.p>
                <styled.p>
                  Participants to any ReactJS Acadmey event that violate these rules may be sanctioned or expelled from the event without a refund at the discretion of the conference organisers.
                </styled.p>
              </Styled(Col)>
            </Row>
          </Styled(Col)>
        </Row>
      </Card>
    </Grid>
  </TopSection>
  <Section>
    <Grid>
      <Row>
        <Styled(Col)
          md={10}
          mdOffset={1}
          xs={12}
        >
          <Styled(styled.h2)>
            More detail
            <Link
              name="more-detail"
              to="#more-detail"
            >
              #
            </Link>
          </Styled(styled.h2)>
        </Styled(Col)>
      </Row>
      <Row>
        <Styled(Col)
          md={5}
          mdOffset={1}
          xs={12}
        >
          <styled.p>
            Harassment includes offensive verbal comments related to gender, gender identity and expression, age, sexual orientation, disability, physical appearance, body size, race, ethnicity, religion, technology choices, sexual images in public spaces, deliberate intimidation, stalking, following, harassing photography or recording, sustained disruption of talks or other events, inappropriate physical contact, and unwelcome sexual attention.
          </styled.p>
          <styled.p>
            Participants asked to stop any harassing behavior are expected to comply immediately.
          </styled.p>
          <styled.p>
            Sponsors and coaches are also subject to the anti-harassment policy. In particular, sponsors and coaches should not use sexualised images, activities, or other material. Additionally, sponsors and coaches should not use sexualised clothing/uniforms/costumes, or otherwise create a sexualised environment.
          </styled.p>
          <styled.p>
            If a participant engages in harassing behavior, the event organisers may take any action they deem appropriate, including warning the offender or expulsion from the event with no refund.
          </styled.p>
          <styled.p>
            If you are being harassed, notice that someone else is being harassed, or have any other concerns, please contact a member of ReactJS Academy staff immediately. Our coaches and representatives will identify themselves to you at the start of any event and may also be identified via branded clothing.
          </styled.p>
          <styled.p>
            ReactJS Academy staff will be happy to help participants contact local law enforcement, source escorts, or otherwise assist those experiencing harassment to feel safe for the duration of the training event. We value your attendance immensly and want you to have the best experience possible.
          </styled.p>
          <styled.p>
            We expect participants to follow these rules at our trainings and training-related social events.
          </styled.p>
        </Styled(Col)>
        <Styled(Col)
          md={5}
          mdOffset={1}
          xs={12}
        >
          <styled.img
            src="https://firebasestorage.googleapis.com/v0/b/reactjsacademy-react.appspot.com/o/codeofconduct_moredetail.jpg?alt=media"
          />
        </Styled(Col)>
      </Row>
    </Grid>
  </Section>
  <UpcomingTrainingSection />
</Fragment>
`;
